lib/common/logging.c

```diff
@@ -239,16 +241,31 @@ crm_add_logfile(const char *filename)
             }
         }
 
+        modestr = getenv("PCMK_logfile_mode");
+        if (modestr) {
+            long filemode_l = strtol(modestr, NULL, 8);
+            if (filemode_l == LONG_MIN || filemode_l == LONG_MAX) {
+	        filemode = 0;
+            } else {
+		    filemode = (mode_t)filemode_l;
+	    }
+        }
+
         if (fix) {
             rc = fchown(logfd, pcmk_uid, pcmk_gid);
             if (rc < 0) {
                 crm_warn("Cannot change the ownership of %s to user %s and gid %d",
                          filename, CRM_DAEMON_USER, pcmk_gid);
             }
+	    if (!filemode) {
+                    filemode = S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP;
```

Comment:
I think you could just initialize filemode to this when you first create the variable and then skip needing an extra block.

Reply:
this avoids having to duplicate the default mask in the strtol() failure case though

Reply:
though defining DEFAULT_LOGFILE_MASK could get around that